src/posts/posts.service.ts

```diff
@@ -35,39 +40,72 @@ export class PostsService {
 
   async createPost(
     postCreationDto: PostCreationDto,
+    userId: number,
   ): Promise<PostCreationInterface> {
-    const createdPost = await this.postRepository.createPost(postCreationDto);
+    const createdPost = await this.postRepository.createPost(
+      postCreationDto,
+      userId,
+    );
     return { id: createdPost.uuid };
   }
 
   async flagPost(
     params: PostIdParam,
     flagPostDto: FlagPostFinishedDto,
   ): Promise<void> {
-    await this.postRepository.flagPostCreation(
-      flagPostDto.finished,
-      params.postid,
-    );
+    // get post
+    const post = await this.postRepository.findPostById(params.postid);
+
+    // check whether post is found
+    if (!post) {
+      throw new NotFoundException(`Post with id: ${params.postid} not found`);
+    }
+
+    await this.postRepository.flagPostCreation(flagPostDto.finished, post);
   }
 
-  async deletePost(postid: string): Promise<void> {
-    await this.postRepository.deletePost(postid);
+  async deletePost(postid: string, userId: number): Promise<void> {
+    // get post
+    const post = await this.postRepository.findPostById(postid);
+
+    // check whether post is found
+    if (!post) {
+      throw new NotFoundException(`Post with id: ${postid} not found`);
+    }
+
+    // Check if current user is the owner of the post
+    if (post.user_id !== userId) {
+      throw new UnauthorizedException('Unauthorized');
+    }
+
+    await this.postRepository.remove(post);
   }
 
   async createOptionGroup(
     postid: string,
     groupsCreationDto: OptionsGroupCreationDto,
   ): Promise<OptionsGroups> {
     const response: OptionsGroups = { groups: [] };
+
+    // get post
+    const post = await this.postRepository.findPostById(postid);
+
+    // check whether post found
+    if (!post) {
+      throw new NotFoundException(`Post with id: ${postid} not found`);
+    }
+
     // Loop through all groups
     for (let i = 0; i < groupsCreationDto.groups.length; i++) {
       const group = groupsCreationDto.groups[i];
       const createdGroup = await this.groupRepository.createGroup(
-        postid,
+        post,
```

Comment:
Why don't you pass the options here to be created with the group creation?

Reply:
@Sahl-A did you check this suggestion?

Reply:
yes working on it.

Reply:
I couldn't create the options with the group creation. However, I refactored the code to create bulk options to simplify the logic and to improve the performance by making 1 database call for multiple options.